test/common/odp_cunit_common.c

```diff
@@ -291,13 +294,30 @@ static int cunit_update_suite(odp_suiteinfo_t *updated_sinfo)
 int odp_cunit_run(void)
 {
 	int ret;
+#ifdef TEST_CUNIT_XML
+	char name[255];
+#endif
 
 	printf("\tODP API version: %s\n", odp_version_api_str());
 	printf("\tODP implementation name:    %s\n", odp_version_impl_name());
 	printf("\tODP implementation version: %s\n", odp_version_impl_str());
 
+#ifdef TEST_CUNIT_XML
+	if (strlen(__progname)) {
+		snprintf(name, 255, "%s", __progname);
+		CU_set_output_filename(name);
+	} else {
+		/* forked processes do not have __progname */
+		ret = readlink("/proc/self/exe", name, sizeof(name) - 1);
+		name[ret] = 0;
+		CU_set_output_filename(basename(name));
+	}
+
+	CU_automated_run_tests();
+#else
```

Comment:
This should be refactored to a function which is selected basing on `TEST_CUNIT_XML`, rather than having `#ifdef` directly in the function body.

Reply:
@muvarov I still do not like this piece. Maybe you can add a runtime switch through environment variable? Then there will be no requirement for extra AM_CONDITIONAL/AC_DEFINE